packages/radix-ui-themes/src/components/radio.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import * as React from 'react';
+import classNames from 'classnames';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { composeRefs } from '@radix-ui/react-compose-refs';
+import { radioPropDefs } from './radio.props.js';
+import { extractProps } from '../helpers/index.js';
+import { marginPropDefs } from '../props/index.js';
+
+import type { ComponentPropsWithoutColor, NotInputRadioAttributes } from '../helpers/index.js';
+import type { MarginProps, GetPropDefTypes } from '../props/index.js';
+
+type RadioElement = React.ElementRef<'input'>;
+type RadioOwnProps = GetPropDefTypes<typeof radioPropDefs> & {
+  value: string;
+  onValueChange?: (value: string) => void;
+};
+type RadioInputProps = Omit<
+  ComponentPropsWithoutColor<'input'>,
+  NotInputRadioAttributes | 'defaultValue' | 'value'
```

Comment:
Why are we removing `value`?

Reply:
It's added just above as a required prop